### ISSUE TITLE
CapturedOutputImpl reports 1 log line if log is empty

### ISSUE THREAD

**Issue by winne42** [joke/spock-outputcapture#157]
Hi @joke , we discovered that `CapturedOutputImpl.getLines()` results in a String-list with one (empty) element if nothing has been logged yet.

We would love if you added a special treatment for the case that nothing has been logged yet:
```groovy
@Override
    List<String> getLines() {
        if (buffer.length() == 0) return []
        return asList(buffer.toString().split("(\\r\\n|\\n|\\r)"));
    }
```

**Comment by winne42** [comment]
Thanks so much, @joke !!

**Comment by winne42** [comment]
Sorry, @joke, while I can confirm that your solution works with Java 8 (coretto-1.8), which is why your pipeline is green it does _not_ work with Java 17 (adoptium-17). The reason seems to be a change in the JDK from Java 8 to 9, see bug report https://bugs.openjdk.org/browse/JDK-8199771 that describes the exact problem and https://bugs.openjdk.org/browse/JDK-8069325 the ticket number under which it was changed.

Long story short: I have the feeling that we need an `if (buffer.length() == 0) return Collections.emptyList();` (or `new ArrayList<>()` if you think a mutable list is more appropriate)

**Comment by joke** [comment]
@winne42 thanks for reporting the issue.
